### PR TITLE
Support -notop option in coqc

### PR DESCRIPTION
**Pull request by elazarg**

### CODE REVIEW

**Comment by maximedenes** [comment]
This is interesting, I mentioned the `-notop` option during the last working group, because it seemed fundamentally flawed to me. Nobody seemed to recall what purpose it was serving. Can you elaborate on what you use this option for?

**Comment by elazarg** [comment]
I use coqide and "compile buffer", and it seems to compile without Top. For
some reason it does not compile one of the files (without any error
message), so I turned to use coqc directly. But without "-notop" it cannot
interoperate with coqide.

(BTW passing arguments to coqc through coq_makefile also seems to be
broken, and so is the "create makefile" option in coqide in Windows)

בתאריך יום ו׳, 23 בדצמ' 2016, 13:34, מאת Maxime Dénès ‏<
notifications@github.com>:

**Comment by maximedenes** [comment]
"Compile Buffer" calls `coqc`. You are saying that `coqc` does not support `-notop`. But you seem to be also saying that "Compile Buffer" passes `-notop` to `coqc`. What am I missing?

**Comment by elazarg** [comment]
I know. And yet, it happens. Running "compile buffer" on one of my files (a slightly modified `Integers.`v from CompCert/lib) creates only a `.glob` file and not `.vo`, although running it on any other file produces `.vo` and running coqc from the command line does create a `.vo` file.

I can't reproduce the issue with `Top` so I can't give you a minimal example, but it happened consistently. (at the end I have edited the `.exe` file with a text editor, and replaced the `-color` option with `-notop`. I wouldn't go for that option if I had another one).

There are many issues with coqide on Windows. They are hard to reproduce so they are hard to report, but until they are resolved I need the workarounds.

**Comment by maximedenes** [comment]
There is a known bug under Windows (#5212) that makes "Compile Buffer" not report compilation errors, which explains the first problem you describe.

I guess the main question I have is: what is `-notop` supposed to do on `coqc`?

**Comment by elazarg** [comment]
It does not explain the problem, since there are no compliation errors. The printed command runs smoothly from the command line. Any error comes from within coqide.

As far as I am concerned, `-notop` supposed to be the magic the prevents coqide from complaining that library X have a name "Top.y" but does not have a name "y" when I try to use "y", and vice versa when compiling from coqide and trying to use it from the command line. It should work without this flag, but it does not do so reliably, so I need that flag.

In other words, it should work (for now) as it does in this PR.

**Comment by ejgallego** [comment]
I think this is wrong, the `-top` family of options are intended to set the module name for interactive sessions. However, in coqc this is set by the module name. Using `-notop` to override that is wrong.

I propose this PR to be closed and a proper bug with a reproducible case to be filled.

**Comment by ejgallego** [comment]
As Maxime has pointed out, the most likely cause for a `.vo` file not to be generated is an error in the input file. I suggest you use `coqide.exe -debug` to provide more information in the bug report.

**Comment by herbelin** [comment]
@elazarg: We closed the PR but don't hesitate to fill a bug report with the problems you have so that the discussion can continue and that a solution can eventually be found.

**Comment by ejgallego** [comment]
So indeed @elazarg closing this PR means that after discussion we think the code should not be merged, but I'm sure there is plenty of interest in trying to solve your problem.

Something that occurred to me just now is that for some people interacting in github is easier than going to Coq's bugzilla and opening an account, etc... I really don't like Bugzilla for discussion because its lack of features.

So I suggest that if you find using Bugzilla uneasy, we use this PR for discussion about your bug.
